c7n/resources/account.py

```diff
@@ -970,3 +970,70 @@ def process(self, resources):
                 if e.response['Error']['Code'] == 'ResourceNotFoundException':
                     return
                 raise
+
+
+@filters.register('xray-encrypt-key')
+class XrayEncrypted(Filter):
+    """Determine if xray is encrypted.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: xray-encrypt-with-default
+                resource: aws.account
+                filters:
+                  - type: xray-encrypt-key
+                    key: default
+              - name: xray-encrypt-with-kms
+                  - type: xray-encrypt-key
+                    key: kms
+    """
+
+    permissions = ('xray:GetEncryptionConfig',)
+    schema = type_schema(
+        'xray-encrypt-key',
+        required=['key'],
+        key={'type': 'string', 'enum': ['default', 'kms']}
+    )
+
+    def process(self, resources, event=None):
+        client = self.manager.session_factory().client('xray')
+        gec_result = client.get_encryption_config()['EncryptionConfig']
+        resources[0]['c7n:XrayEncryptionConfig'] = gec_result
+        kv = 'KMS' if self.data.get('key') == 'kms' else 'NONE'
+        return resources if (gec_result['Type'] == kv) else []
+
+
+@actions.register('set-xray-encrypt')
+class SetXrayEncryption(BaseAction):
+    """Enable specific xray encryption.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: xray-default-encrypt
+                resource: aws.account
+                actions:
+                  - type: set-xray-encrypt
+                    key: default
+              - name: xray-kms-encrypt
+                  - type: set-xray-encrypt
+                    key: alias/some/alias/ke
```

Comment:
typically services using kms would require a keyid, not an alias, if we're going to support aliases here we need to do some additional work to resolve the alias to a keyid.

Reply:
chuck pointed out that xray api does support aliases on set

Reply:
Updated the comments as you suggested and changed default to absent in set-xray-encrypt.